### PR TITLE
added solution to binary search

### PR DESCRIPTION
**Pull request by haitle16**
- [X] Top-level README “Table of Contents” is updated
 - [X] Feature tasks for this challenge are completed
 - [X] Unit tests written and passing
     - [X] “Happy Path” - Expected outcome
     - [X] Expected failure
     - [ ] Edge Case (if applicable/obvious)
 - [X] README for this challenge is complete
     - [X] Summary, Description, Approach & Efficiency, Solution
     - [X] Link to code
     - [X] Picture of whiteboard

### CODE REVIEW

File: java401challenges/src/test/java/java401challenges/BinarySearchTest.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    @Test public void testBinarySearch() {
+        int result = 2;
+        int actual = BinarySearch.binSearch(new int[] {2,5,7,8,10,12,14}, 7);
+        int result2 = 5;
+        int actual2 = BinarySearch.binSearch(new int[] {2,5,7,8,10,12,14}, 12);
+        int result3 = -1;
+        int actual3 = BinarySearch.binSearch(new int[] {}, 12);
+        assertEquals( actual, result);
+        assertEquals( actual2, result2);
+        assertEquals( actual3, result3);
```
Comment (Zaffierce) [inline java401challenges/src/test/java/java401challenges/BinarySearchTest.java:16]: I like the idea of the three tests, but each test should have it's own method.
